src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -347,7 +347,17 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
                 // This is required for Bitbucket Server to update the refs/pull-requests/* references
                 // See https://community.atlassian.com/t5/Bitbucket-questions/Change-pull-request-refs-after-Commit-instead-of-after-Approval/qaq-p/194702#M6829
                 if (endpoint.isCallCanMerge()) {
-                    pullRequest.setCanMerge(getPullRequestCanMergeById(pullRequest.getId()));
+                    try {
```

Comment:
It will handle 409, but does it really need to iterate on **all** open pull requests when building single pull request?

Reply:
I agree but it is somewhere else in the code :). 
This PR is a very low level change regarding handling the status without throwing an Exception.